src/panels/lovelace/cards/hui-conditional-card.ts

```diff
@@ -1,70 +1,88 @@
-import computeCardSize from "../common/compute-card-size.js";
-import createCardElement from "../common/create-card-element.js";
-import { HomeAssistant } from "../../../types.js";
-import { LovelaceCard, LovelaceConfig } from "../types.js";
-
-interface Condition {
-  entity: string;
-  state?: string;
-  state_not?: string;
-}
-
-interface Config extends LovelaceConfig {
-  card: LovelaceConfig;
-  conditions: Condition[];
-}
-
-class HuiConditionalCard extends HTMLElement implements LovelaceCard {
-  protected config?: Config;
-  protected card?: LovelaceCard;
-
-  public setConfig(config) {
-    if (
-      !config.card ||
-      !config.conditions ||
-      !Array.isArray(config.conditions) ||
-      !config.conditions.every((c) => c.entity && (c.state || c.state_not))
-    ) {
-      throw new Error("Error in card configuration.");
-    }
-
-    this.config = config;
-    this.card = createCardElement(config.card);
-    if (this.card) {
-      this.appendChild(this.card);
-      this.card.hass = this.hass;
-    }
-  }
-
-  set hass(hass: HomeAssistant) {
-    const visible =
-      this.config &&
-      this.config.conditions.every((c) => {
-        if (!(c.entity in hass.states)) {
-          return false;
-        }
-        if (c.state) {
-          return hass.states[c.entity].state === c.state;
-        }
-        return hass.states[c.entity].state !== c.state_not;
-      });
-
-    if (visible && this.card) {
-      this.card.hass = hass;
-    }
-
-    this.style.setProperty("display", visible ? "" : "none");
-  }
-
-  public getCardSize() {
-    return computeCardSize(this.card);
-  }
-}
-
-declare global {
-  interface HTMLElementTagNameMap {
-    "hui-conditional-card": HuiConditionalCard;
-  }
-}
-
-customElements.define("hui-conditional-card", HuiConditionalCard);
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+
+import computeCardSize from "../common/compute-card-size.js";
+import createCardElement from "../common/create-card-element.js";
+
+import { HomeAssistant } from "../../../types.js";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+interface Condition {
+  entity: string;
+  state?: string;
+  state_not?: string;
+}
+
+interface Config extends LovelaceConfig {
+  card: LovelaceConfig;
+  conditions: Condition[];
+}
+
+class HuiConditionalCard extends LitElement implements LovelaceCard {

```

Comment:
I don't understand, why are you converting this to `LitElement` ? That's not needed ?

Reply:
I thought that was something we were doing?

Reply:
we are converting from Polymer Element to Lit Element. This element was an HTML Element, which is very basic with no templating etc, but that's fine because the card only shows/hides an element based on a condition.

Reply:
Is there a good rule of thumb on when we should use Lit vs plain HTML?